include/deal.II/base/mpi_consensus_algorithms.templates.h

```diff
@@ -502,27 +503,35 @@ namespace Utilities
         const int tag_deliver = Utilities::MPI::internal::Tags::
           consensus_algorithm_pex_process_deliver;
 
+        // Wait until we have a message ready for retrieval, though we don't
+        // care which process it is from. We know that the source must be
+        // listed in the 'sources' array, though.
         MPI_Status status;
-        auto ierr = MPI_Probe(MPI_ANY_SOURCE, tag_request, this->comm, &status);
+        int ierr = MPI_Probe(MPI_ANY_SOURCE, tag_request, this->comm, &status);
```

Comment:
We always write `auto ierr`. Let's keep this way.

Reply:
I think that's debatable. We have
```
> grep -r MPI_ source/ | grep ierr | grep auto | wc -l
2
> grep -r MPI_ source/ | grep ierr | grep int | wc -l
89
```
So pretty universally at least in those places where the return type shows up on the same line as the MPI call, we actually do spell it out.

If you'd like me to, I promise to make it uniform in this file once all of the patches I have already sitting around are merged (too many merge conflicts otherwise).